cloud/services/object_storage_objects_test.go

```diff
@@ -0,0 +1,287 @@
+package services
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	awssigner "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/go-logr/logr/testr"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	infrav1alpha2 "github.com/linode/cluster-api-provider-linode/api/v1alpha2"
+	"github.com/linode/cluster-api-provider-linode/cloud/scope"
+	"github.com/linode/cluster-api-provider-linode/mock"
+
+	. "github.com/linode/cluster-api-provider-linode/mock/mocktest"
+)
+
+func TestCreateObject(t *testing.T) {
+	t.Parallel()
+
+	var log = testr.New(t)
+
+	NewSuite(t, mock.MockK8sClient{}, mock.MockS3Client{}, mock.MockS3PresignClient{}).Run(
+		OneOf(
+			Path(
+				Result("nil machine scope", func(ctx context.Context, mck Mock) {
+					_, err := CreateObject(ctx, nil, []byte("fake"), log)
+					assert.Error(t, err)
+				}),
+				Result("nil s3 client", func(ctx context.Context, mck Mock) {
+					_, err := CreateObject(ctx, &scope.MachineScope{
+						LinodeCluster: &infrav1alpha2.LinodeCluster{
+							Spec: infrav1alpha2.LinodeClusterSpec{
+								ObjectStore: &infrav1alpha2.ObjectStore{
+									CredentialsRef: corev1.SecretReference{Name: "fake"}}}}},
+						[]byte("fake"), log)
+					assert.Error(t, err)
```

Comment:
When defining the errors as variables from https://github.com/linode/cluster-api-provider-linode/pull/586/files#r1876417132, we should also check the error is the expected error